assembly/src/assembler/instruction/ext2_ops.rs

```diff
@@ -52,8 +53,11 @@ pub fn ext2_mul(span: &mut BasicBlockBuilder) {
 /// operations outputs the result c = (c1, c0) where c = a * b^-1.
 ///
 /// This operation takes 11 VM cycles.
-pub fn ext2_div(span: &mut BasicBlockBuilder) {
-    span.push_advice_injector(Ext2Inv);
+pub fn ext2_div(
+    span: &mut BasicBlockBuilder,
+    mast_forest_builder: &mut MastForestBuilder,
+) -> Result<(), AssemblyError> {
```

Comment:
Good idea! Implemented.

The minor downside is that it makes `Assembler::compile_body()` slightly less clean, since `basic_block_builder` owns the `MastForestBuilder` mutable reference [here](https://github.com/0xPolygonMiden/miden-vm/blob/cd6fa22a1cffff5e54b77f4b4672bd7816eaed10/assembly/src/assembler/mod.rs#L603), and so we need to access the `MastForestBuilder` through it in the rest of the method. But it makes the rest of the codebase much cleaner, so IMO it's still definitely worth it.

Reply:
Agreed! One nit (feel free to ignore): I noticed that in some cases we've renamed `span` variable into `basic_block_builder` variable. I would probably have shortened it to just `block_builder` or maybe even just `block`.